iosHackthonApp/Modules/Features/SignUp/Sources/SignUp/SignUpWithEmail.swift

```diff
@@ -9,22 +9,32 @@
 import SwiftUI
 import Strings
 import Components
+import shared
 
 struct SignUpWithEmail: View {
 	@State private var email: String = ""
-	
+	@State private var moveToNextScreen: Bool = false
+
 	var body: some View {
 		VStack(alignment: .leading) {
 			SignUpInfoView(title: Strings.SignUp.signUpWithEmailTitle, description: Strings.SignUp.signUpWithEmailInfo)
 
 			Spacer().frame(maxHeight: .infinity)
 			
 			GeometryReader { geometry in
-				SignUpUserEntryView(isSignupWithEmail: true,
-									textFieldPlaceholder: Strings.SignUp.signUpWithEmailPlaceHolder, buttonTitle: Strings.SignUp.signUpWithEmailButtonTitle,
-									width: geometry.size.width,
-									destinationView: SignUpWithPassword(),
-									entryField: $email)
+				let viewModel = SignUpUserEntryViewModel(
+					isSignupWithEmail: true,
+					textFieldPlaceholder: Strings.SignUp.signUpWithEmailPlaceHolder,
+					buttonTitle: Strings.SignUp.signUpWithEmailButtonTitle,
```

Comment:
Could we set these directly in the viewModel if they are constants?

Reply:
moved to viewmodels